bottlerocket-settings-models/settings-extensions/image-verification-plugins/src/lib.rs

```diff
@@ -0,0 +1,102 @@
+//! Settings related to Image Verification Plugins
+
+use bottlerocket_model_derive::model;
+use bottlerocket_modeled_types::ValidBase64;
+use bottlerocket_settings_sdk::{GenerateResult, SettingsModel};
+use std::convert::Infallible;
+
+#[model(impl_default = true)]
+pub struct ImageVerificationPluginsSettingsV1 {
+    notation: NotationSettings,
+}
+
+#[model(impl_default = true)]
+pub struct NotationSettings {
+    /// Base64 encoded trustpolicy.json
+    /// https://github.com/notaryproject/specifications/blob/main/specs/trust-store-trust-policy.md#trust-store
+    trustpolicy: ValidBase64,
+}
+
+type Result<T> = std::result::Result<T, Infallible>;
+
+impl SettingsModel for ImageVerificationPluginsSettingsV1 {
+    type PartialKind = Self;
+    type ErrorKind = Infallible;
+
+    fn get_version() -> &'static str {
+        "v1"
+    }
+
+    fn set(_current_value: Option<Self>, _target: Self) -> Result<()> {
+        // Set anything that can be parsed as ImageVerificationPluginsSettingsV1.
+        Ok(())
+    }
+
+    fn generate(
+        existing_partial: Option<Self::PartialKind>,
+        _dependent_settings: Option<serde_json::Value>,
+    ) -> Result<GenerateResult<Self::PartialKind, Self>> {
+        Ok(GenerateResult::Complete(
+            existing_partial.unwrap_or_default(),
+        ))
+    }
+
+    fn validate(_value: Self, _validated_settings: Option<serde_json::Value>) -> Result<()> {
+        // ImageVerificationPluginsSettingsV1 is validated during deserialization.
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use serde_json::json;
+
+    #[test]
+    fn test_generate_image_verification_plugins_settings() {
+        assert_eq!(
+            ImageVerificationPluginsSettingsV1::generate(None, None),
+            Ok(GenerateResult::Complete(
+                ImageVerificationPluginsSettingsV1 { notation: None }
+            ))
+        )
+    }
+
+    #[test]
+    fn test_serde_image_verification_plugins() {
+        let test_json = json!({
+            "notation": {
+                "trustpolicy": "ewogICJ2ZXJzaW9uIjogIjEuMCIsCiAgInRydXN0UG9saWNpZXMiOiBbXQp9"
+            }
+        });
+
+        let test_json_str = test_json.to_string();
+
+        let image_verification_plugins_settings: ImageVerificationPluginsSettingsV1 =
+            serde_json::from_str(&test_json_str).unwrap();
+
+        assert_eq!(
+            image_verification_plugins_settings
+                .notation
+                .as_ref()
+                .unwrap()
+                .trustpolicy
+                .as_ref()
+                .unwrap()
+                .as_ref(),
+            "ewogICJ2ZXJzaW9uIjogIjEuMCIsCiAgInRydXN0UG9saWNpZXMiOiBbXQp9"
+        );
+    }
```

Comment:
Nit: probably a bit redundant, do you think we should add a negative case here when the input base64 is invalid?